src/components/kpi/kpi.tmpl.html

```diff
@@ -61,8 +61,16 @@
 
       </div>
     </div>
+    <!-- Data not found display -->
+    <div class="kpi-no-data" ng-if="isDataNotFound()">
+      <!-- Remove KPI -->
+      <button type="button" ng-show="isEditing()" class="kpi-remove im-fab im-warn" ng-click="deleteKpi()">
+        <i class="fa fa-times" aria-hidden="true"></i>
+      </button>
+      <div common-data-not-found is-kpi="true" widget-engine="::kpi.endpoint"/>
```

Comment:
warning there I think you've renamed `widget-engine` in the directive. Don't you have to update widget.tmpl.html as well ?